staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -607,6 +609,50 @@ func TestDetectSafeNotMountedBehavior(t *testing.T) {
 	}
 }
 
+func TestCheckUmountError(t *testing.T) {
+	target := "/test/path"
+	withSafeNotMountedBehavior := true
+
+	pState := &os.ProcessState{}
+	var ref_pState reflect.Value = reflect.ValueOf(pState).Elem()
+	var ref_status reflect.Value = ref_pState.FieldByName("status")
+	status := (*uint32)(unsafe.Pointer(ref_status.UnsafeAddr()))
```

Comment:
using unsafe to fiddle with unexported fields is surprising to me... is there not a way we can stub this out instead?

Reply:
`TestCheckUmountError()` tests `checkUmountError()` func behavior which checks the result of `umount` command(callled in `Unmount()`, `tryUnmount()` and `forceUmount()`) .
In this test case, `unmount` command is not actually executed so dummy result(`*exec.Cmd` including `ProcessState.status=0`) is needed as `checkUmountError()` func arg.
I couldn't think of any other way than creating `*exec.Cmd` instance directly...
If you know a better way, I would appreciate it if you could tell me.

Reply:
I'm not sure the fragility of this test (depending on unexported fields in the go error types via unsafe/reflect) is worth the improved coverage... this is essentially a reverse implementation of the detection of `checkUmountError`, rather than a check that that function is actually checking the right things

Reply:
`checkUmountError()` needs some `*exec.Cmd` instance with `ProcessState.status=0` for testing.
Another way alternative to the above is to generate it by executing some command(ex. `uname -r`) that returns `status 0` as a stub instead of fiddling unexported fields.

```go
	command := exec.Command("uname", "-r") // dummy command return status 0
	if _, err := command.CombinedOutput(); err != nil {
		t.Errorf("Faildto exec dummy command . err: %s", err)
	}
```

How about this?

Reply:
In addition, I've considered to use [testingexec package](https://pkg.go.dev/k8s.io/utils/exec/testing) for dummy command.
However, [FakeCmd](https://pkg.go.dev/k8s.io/utils/exec/testing#FakeCmd) dose not seem to have field containing status code(correspond to [exec.Cmd.ProcessState](https://pkg.go.dev/os/exec#Cmd.ProcessState)).

Reply:
I'll let @jsafrane decide if this is a reasonable way to test the function

Reply:
@jsafrane @liggitt 

In this PR, I included two error check logic in `checkUmountError()`.

- `errNoChildProcesses` check
  - `no child processes` error handling logic for `umount` command and `ProcessState.status` will be checked in it
  - Implemented in #103780 and [the unit test was not implemented previously](https://github.com/kubernetes/kubernetes/pull/103780#discussion_r673550831)
  - Essentially out of scope for this fix
- `errNotMounted` check
  - `not mounted` error handling logic for `umount` command
  - The main target of this fix and unit test

We need  a `*exec.Cmd` instance with `ProcessState.status=0` for unit test because of `errNoChildProcesses` check.
Therefore, if necessary, I'll consider separating the `errNoChildProcesses` check from the `checkUmountError()`. (However, in this case, the same `errNoChildProcesses` check logic must be written three times in each functions (`Unmount()`, `tryUnmount()`, `forceUmount()`)).

Reply:
@jsafrane thx!